### PR TITLE
Remove TransactionRuntimeException

### PR DESCRIPTION
**Pull request by brfrn169**
This is a refactoring thing. Please take a look!

### CODE REVIEW

**Review by thongdk8 — DISMISSED** [review]
LGTM! Thank you!

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/Coordinator.java [hunk 1]
```diff
@@ -185,15 +184,15 @@ public int hashCode() {
     private void checkNotMissingRequired(Result result) {
       if (!result.getValue(Attribute.ID).isPresent()
           || !result.getValue(Attribute.ID).get().getAsString().isPresent()) {
-        throw new RequiredValueMissingException("id is missing in the coordinator state");
+        throw new IllegalArgumentException("id is missing in the coordinator state");
```
Comment (feeblefakie) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/Coordinator.java:187]: I'm not sure if `IllegalArgumentException` is appropriate here. It is more like a software's bug, so something different should be used?

**Review by feeblefakie — COMMENTED** [review]
Overall looking good! Left one question.

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/Coordinator.java (hunk 1 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/transaction/consensuscommit/Coordinator.java:187]: Thank you for taking a look at this!

Yes, that's true. We should not use `IllegalArgumentException` here. Maybe we can simply use `CoordinatorException` here. I'll fix it. Thanks.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**